### PR TITLE
Add --port option to textual console.

### LINKED ISSUE #2014: Allow command-line option to change development port
I couldn't use `textual run --dev` because the server I was on had AV software listening on port 8081.

Luckily a quick `git grep` was able to fix it, but it would be nice to have that as a command-line option.

### PR DESCRIPTION
**Pull request by rodrigogiraoserrao**
**Please review the following checklist.**

- [x] Docstrings on all new or modified functions / classes 
- [x] Updated documentation
- [x] Updated CHANGELOG.md (where appropriate)

This will close #2014

### CODE REVIEW

File: src/textual/app.py [hunk 1]
```diff
@@ -58,6 +58,8 @@
 from rich.segment import Segment, Segments
 from rich.traceback import Traceback
 
+from textual.devtools.client import get_port_for_devtools
```
Comment (willmcgugan) [inline src/textual/app.py:61]: Can you use relative imports please.

File: src/textual/cli/cli.py [hunk 2]
```diff
@@ -21,14 +23,27 @@ def run():
 
 
 @run.command(help="Run the Textual Devtools console.")
+@click.option(
+    "--port",
+    "port",
+    type=int,
+    default=None,
```
Comment (willmcgugan) [inline src/textual/cli/cli.py:30]: Can the default not be an integer? So it shows in the help.

File: src/textual/devtools/client.py [hunk 3]
```diff
@@ -17,11 +18,20 @@
...
 from .._log import LogGroup, LogVerbosity
 
-DEVTOOLS_PORT = 8081
+DEVTOOLS_PORT_ENVIRON_VARIABLE = "TEXTUAL_CONSOLE_PORT"
+DEFAULT_DEVTOOLS_PORT = 8081
 WEBSOCKET_CONNECT_TIMEOUT = 3
 LOG_QUEUE_MAXSIZE = 512
 
 
+def get_port_for_devtools() -> int:
```
Comment (willmcgugan) [inline src/textual/devtools/client.py:27]: Can you move this to `constants.py` where we read environment variables.

File: src/textual/cli/cli.py (hunk 2 above)
Comment (rodrigogiraoserrao) [reply src/textual/cli/cli.py:30]: I don't understand. The default is _not_ an integer (it is `None`).
If I change the default to an integer, it still doesn't show when I run `textual console --help`.

Comment (willmcgugan) [reply src/textual/cli/cli.py:30]: I thought it would show the default. That's disappointing. In that case, would you mind updating the help text with the default?

Comment (rodrigogiraoserrao) [reply src/textual/cli/cli.py:30]: 👍

File: src/textual/constants.py [hunk 4]
```diff
@@ -25,10 +28,18 @@ def get_environ_bool(name: str) -> bool:
     Returns:
         `True` if the env var is "1", otherwise `False`.
     """
-    has_environ = os.environ.get(name) == "1"
+    has_environ = get_environ(name) == "1"
     return has_environ
 
 
+def get_port_for_devtools() -> int:
```
Comment (willmcgugan) [inline src/textual/constants.py:35]: This doesn't need to be a callable, as it is never going to change.

Can you create a constant, in the same way as the other symbols in this module.

**Review by willmcgugan — CHANGES_REQUESTED** [review]

File: src/textual/constants.py (hunk 4 above)
Comment (rodrigogiraoserrao) [reply src/textual/constants.py]: I thought a callable made sense because we had to check if we'd use the default, or not.
Let me try again.

File: src/textual/constants.py [hunk 5]
```diff
@@ -37,3 +37,12 @@ def get_environ_bool(name: str) -> bool:
 
 LOG_FILE: Final[str | None] = get_environ("TEXTUAL_LOG", None)
 """A last resort log file that appends all logs, when devtools isn't working."""
+
+
+DEVTOOLS_PORT_ENVIRON_VARIABLE = "TEXTUAL_CONSOLE_PORT"
+DEFAULT_DEVTOOLS_PORT = 8081
+try:
```
Comment (willmcgugan) [inline src/textual/constants.py:44]: Let's factor this out in to a function. Something like `get_environment_int(env_var, default)`

**Review by willmcgugan — CHANGES_REQUESTED** [review]

File: src/textual/constants.py (hunk 5 above)
Comment (rodrigogiraoserrao) [reply src/textual/constants.py]: I pondered whether the `default` could be set to `None` (to return `None`) if the env var wasn't set but then decided against it because it would take an `overload` to ensure consistent typing and that seemed like overkill for this situation.

**Review by willmcgugan — APPROVED** [review]

File: src/textual/constants.py (hunk 5 above)
Comment (willmcgugan) [reply src/textual/constants.py]: I agree. These constants should have simple typing.
